Added random pilot name generator

maxwellpark opened the pull request:
Method added to the Crew class that generates a random name of a gender based on the argument passed. This could be used when hiring new pilots.

Human name format: Firstname + Initial.
Robot name format: Alphabetical prefix + numerical suffix.

Added robot serial number prefix length and suffix length attributes, so we can vary the format of the serial number if we wish to. Perhaps different kinds of robots have different kinds of serial number.

Created buttons in the Crew panel for demo purposes.

maxwellpark committed:
Added random pilot name generator

Method added to the Crew class that generates a random name of a gender based on the argument passed. This could be used when hiring new pilots.

Human name format: Firstname + Initial.
Robot name format: Alphabetical prefix + numerical suffix.

Added robot serial number prefix length and suffix length attributes, so we can vary the format of the serial number if we wish to. Perhaps different kinds of robots have different kinds of serial number.

Created buttons in the Crew panel for demo purposes.

ambid17 commented on Assets/Scripts/Crew.cs:
```diff
@@ -0,0 +1,70 @@
+		randomRobotNameButton.onClick.AddListener(delegate { randomPilotName("robot"); });
+	}
+
+	private void randomPilotName(string gender)
```
use an enum for things with finite states

ambid17 commented on Assets/Scripts/Crew.cs:
```diff
@@ -0,0 +1,70 @@
+
+public class Crew : MonoBehaviour
+{
+	private string[] maleNames = File.ReadAllLines("./Assets/ImportedAssets/male_names.txt");
```
maybe move these to a singleton or something that cant be run twice? If we end up loading a different scene and come back, they will have to reload the files

maxwellpark committed:
Name generator now accepts an enum instead of a string. Name data moved to Singleton.

ambid17 commented on Assets/Scripts/Crew.cs:
```diff
@@ -24,30 +26,32 @@ public class Crew : MonoBehaviour
+		maleNames = PilotNameDataSingleton.Instance.MaleNames;
+		femaleNames = PilotNameDataSingleton.Instance.FemaleNames;
+
+		randomMaleNameButton.onClick.AddListener(delegate { randomPilotName(Gender.male); });
```
to make sure you cant add two listeners on accident, always call this first: Button.onClick.RemoveAllListeners()

ambid17 commented on Assets/Scripts/Crew.cs:
```diff
@@ -7,9 +7,11 @@
 {
-	private string[] maleNames = File.ReadAllLines("./Assets/ImportedAssets/male_names.txt");
-	private string[] femaleNames = File.ReadAllLines("./Assets/ImportedAssets/female_names.txt");
+	private enum Gender { male, female, robot };
```
enum values are typically capitalized

maxwellpark committed:
Fixed listeners and enum names

Had a null reference exception for the Singleton.Instance. I think it was because the Awake() method of the Crew script was being called before the Awake() method of the Singleton (so the txt file hadn't been read in by the time the names field was being assigned in Crew). So I changed the Crew one to Start(). Seems to have fixed it but not sure if this is a good practice.
